### ISSUE TITLE
User deletion impossible

### ISSUE THREAD

**Issue by salim-b** [formbricks/formbricks#4376]
### Issue Summary

1. Create a new self-hosted Formbricks instance (community edition) and create your default admin user.
2. Invite a new user to your team (only role `owner` is possible in community edition).
3. Make sure the new user finishes their account creation.
4. From your admin user, try to delete the new user.

   Nothing happens and the following error message is displayed on the server:

    ````
    SERVER ERROR:  o [ValidationError]: You cannot delete the last owner of the organization
        at U (/home/nextjs/apps/web/.next/server/app/(app)/environments/[environmentId]/settings/(organization)/general/page.js:1:86158)
        at async S (/home/nextjs/apps/web/.next/server/chunks/5238.js:214:14008)
        at async next (/home/nextjs/apps/web/.next/server/chunks/5238.js:214:13049)
        at async S (/home/nextjs/apps/web/.next/server/chunks/5238.js:214:12465)
        at async next (/home/nextjs/apps/web/.next/server/chunks/5238.js:214:13049)
        at async S (/home/nextjs/apps/web/.next/server/chunks/5238.js:214:12465)
        at async /home/nextjs/apps/web/.next/server/chunks/5238.js:214:14724
        at async /home/nextjs/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:16:418
        at async rE (/home/nextjs/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:15:8148)
        at async r7 (/home/nextjs/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:18:1144) {
      statusCode: 400
    ````

   Screencast:

   https://github.com/user-attachments/assets/aa451cb2-2f7e-4f9d-90be-1ef4491bb4d3

   This happens when trying to delete *any* user.

### Expected Behavior

Step 4\. above should work instead of throwing an error.

### Other information (incl. screenshots, Formbricks version, steps to reproduce,...)

There was a similar issue in the past: https://github.com/formbricks/formbricks/issues/2044

What works as a workaround, is deleting the user from Formbricks PostgreSQL DB (`Users` table) and restarting the container.

### Your Environment

Self-hosted Formbricks

**Comment by gupta-piyush19** [comment]
Hey! @salim-b, thanks for raising this issue. Please try doing it now.
This has already been fixed in some other PR and works fine for me. Please close this ticket if it's fixed.

**Comment by gupta-piyush19** [comment]
Hey! I just rechecked this, so it's in our current release but is working fine on the cloud. It will be fixed in our upcoming release very soon. I'll quickly check today if we can do a patch release with other fixes

**Comment by salim-b** [comment]
@gupta-piyush19 Thank you for looking into this! I can confirm that this issue is resolved in Formbricks v2.7.2.

But there's a (presumedly closely related) critical issue that's still unresolved: https://github.com/formbricks/formbricks/issues/4419
